ui/pages/confirmations/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -431,6 +430,19 @@ export default class ConfirmTransactionBase extends Component {
       );
     };
 
+    const handleNonceChange = ({ target: { value } }) => {
+      const inputValue = Number(value);
+
+      if (inputValue < 0 || isNaN(inputValue)) {
+        updateCustomNonce('');
+        return;
+      }
+
+      updateCustomNonce(String(inputValue));
```

Comment:
Assuming the `inputValue` is a decimal and not hexadecimal, why are we storing it as a string?

Reply:
Good point, no need to store it as a string, removed.